lib/net/dhcp/record.rb

```diff
@@ -75,9 +76,6 @@ def ==(other)
         to_compare << :hostname if other.attrs[:hostname].present? && attrs[:hostname].present?
       end
 
-      # Not all DHCP smart-proxy providers support TFTP filename option (e.g. libvirt).
```

Comment:
Might still be nice to compare these to know if there would be a change. I don't know if we do a `old.record != record && update_record(record)`. Probably better to not rely on this method at all in `conflicts`.

Reply:
I am not following, it is incorrect to include filename in comparsion as it cannot generate conflicts in DHCP servers. The comment was just statement about filename option which is not very relevant to the problem.

Reply:
This method tests for equality. This means that the filename is no longer considered in testing whether 2 records are equal. If you have code that checks if an update is needed by doing `if old.record != record ; update_record(record) ; end` then this change means you will miss updates. That's why I was suggesting that the `conflicts` method shouldn't rely on equality comparison.

Reply:
I haven't noticed it in the codebase, but maybe I missed that and make sense.